### PR TITLE
Constexpr checks in CUDAAllocatorConfig::roundup_power2_divisions

### PR DESCRIPTION
**Pull request by cyyever**
This PR adds constexpr modifier to some template functions to make sure that the checks in CUDAAllocatorConfig::roundup_power2_divisions become compile-time computations.

### CODE REVIEW

File: c10/util/llvmMathExtras.h [hunk 1]
```diff
@@ -147,7 +147,7 @@ std::size_t countTrailingZeros(T Val, ZeroBehavior ZB = ZB_Width) {
 namespace detail {
 template <typename T, std::size_t SizeOfT>
 struct LeadingZerosCounter {
-  static std::size_t count(T Val, ZeroBehavior) {
+  constexpr static std::size_t count(T Val, ZeroBehavior) {
```
Comment (Skylion007) [inline c10/util/llvmMathExtras.h:150]: I think these utilities are from LLVM and we should probably try to either keep it in sync with upstream or make a change log so we can update it later.

Comment (cyyever) [reply c10/util/llvmMathExtras.h:153]: I prefer adding a change log.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Windows fail is real

**Comment by cyyever** [comment]
> Windows fail is real

Unfortunately, windows relies on some platform specific functions which are not constexpr. Now one solution is distinguish between Windows and other platforms by macros in roundup_power2_divisions, which is ugly, and another solution is use std::is_constant_evaluated which needs C++20.

**Comment by ezyang** [comment]
If you are dead set on landing this I would do macros lol

**Pull request closed without merging**